src/main/scala/is/hail/annotations/aggregators/RegionValueTakeAggregator.scala

```diff
@@ -0,0 +1,206 @@
+package is.hail.annotations.aggregators
+
+import is.hail.annotations._
+import is.hail.asm4s._
+import is.hail.expr._
+import is.hail.utils._
+
+object RegionValueTakeBooleanAggregator {
+  def stagedNew(n: Code[Int], mn: Code[Boolean]): Code[RegionValueTakeBooleanAggregator] =
+    (mn).mux(
+      Code._throw(Code.newInstance[RuntimeException, String]("Take-aggregator cannot take NA for length")),
+      Code.newInstance[RegionValueTakeBooleanAggregator, Int](n))
+}
+
+class RegionValueTakeBooleanAggregator(n: Int) extends RegionValueAggregator {
+  private val ab = new MissingBooleanArrayBuilder()
+
+  def seqOp(x: Boolean, missing: Boolean) {
+    if (ab.length() < n)
+      if (missing)
+        ab.addMissing()
+      else
+        ab.add(x)
+  }
+
+  def seqOp(region: Region, off: Long, missing: Boolean) {
+    seqOp(region.loadBoolean(off), missing)
+  }
+
+  def combOp(agg2: RegionValueAggregator) {
+    val that = agg2.asInstanceOf[RegionValueTakeBooleanAggregator]
+    that.ab.foreach { _ =>
```

Comment:
Ah, I see, you can't do that as implemented because you can't support apply(i: Int) because you only store the non-missing values.  Trade-offs, trade-offs.